### ISSUE TITLE
parameter hide / unhide  or active/ inactive or enable/ disable

### ISSUE THREAD

**Issue by sriramab** [gama-platform/gama#2411]
Hi

In continuation with discussion on the forum about making some parameter text locked. I wonder if it is possible to make a parameter active/ inactive based on the parameter chosen before the current parameter. There is already a on_chage: facet, which allows a block of statements to be executed. May be in this block we could say parameter1............. active:false

The idea is to enable/ disable parameters based on a  previously chosen parameter. 

Not a priority at all.

Thank you
Srirama

**Comment by AlexisDrogoul** [comment]
Hi, the option is already available (internally) for preferences in Java. There, you can define whether a bool preference will activate/deactivate other ones, by passing a list of id. For parameters, the only trouble is to identify correctly the activable/disactivable parameters: one might require to pass their full name, which is maybe not very practical if it is a long string...

**Comment by sriramab** [comment]
May be the parameter statement can have an `id:` facet?

**Comment by sriramab** [comment]
Sorry I did not understand your statement: `"the option is already available (internally) for preferences in Java. There, you can define whether a bool preference will activate/deactivate other ones, by passing a list of id" ` ; is it something I can do from GAML?

**Comment by AlexisDrogoul** [comment]
No, you can't. It's purely Java. However, since preferences are internally (in Java) equivalent to parameters (as they are parameters to manipulate the attributes of the `gama` agent), it should be possible to add a facet to parameters so that they can activate or deactivate others. I will be thinking of a solution...
